### PR TITLE
Change configuration file to charon.yaml

### PR DESCRIPTION
**Pull request by ligangty**
New format of charon config file will be used:
* .ini format will be changed to yaml format
* Add targets to decide the target bucket and prefix

Some changes in command line
* Remove --bucket(-b) option from command line
* Add --target(-t) option

### CODE REVIEW

File: charon/config.py [hunk 1]
```diff
@@ -13,68 +13,69 @@
...
-            return default
-        return section[key] if section and key in section else default
+    def get_bucket_prefix(self, target: str) -> str:
+        target_: Dict = self.__targets.get(target, None)
+        if not target_ or not isinstance(target_, Dict):
+            logger.error("The target %s is not found in charon "
+                         "configuration.", target)
+            return None
+        prefix = target_.get("prefix", None)
+        if not prefix:
```
Comment (jdcasey) [inline charon/config.py:60]: It seems like this is a valid state. In this case, the bucket prefix path would be just `/`.

Comment (jdcasey) [reply charon/config.py:60]: Also, I've been through all the changes in this PR so far, and I can't find where get_bucket_prefix() is used. Am I missing it, or has this not yet been implemented?

**Review by jdcasey — CHANGES_REQUESTED** [review]
I mentioned a case where the prefix for a target is None, and we might want to default to the root of the bucket (not append a prefix)...however it's reporting an error in this case. Also, I can't see where the bucket prefix is actually used.

File: charon/config.py (hunk 1 above)
Comment (ligangty) [reply charon/config.py:60]: Yes, the prefix appending function will be used in another PR. This pr is just addressing the config changes. 
And I've set the prefix to empty if it is not specified in config in new commit.

**Review by jdcasey — APPROVED** [review]

**Pull request merged**